ui/jobs/jobs.js

```diff
@@ -1579,7 +1607,35 @@ class Bars {
       this.o.healthBar.fg = computeBackgroundColorFrom(this.o.healthBar, 'hp-color');
   }
 
+  UpdateMPTicker() {
+    if (!this.o.mpTicker) return;
+
+    // Hide out of combat if requested
+    if (!this.options.ShowMPTickerOutOfCombat && !this.inCombat) {
+      this.o.mpTicker.duration = 0;
+      return;
+    }
+
+    let delta = this.mp - this.prevMP;
+    this.prevMP = this.mp;
+    if (!delta)
+      return;
+
+    let baseTick = this.inCombat ? kMPCombatRate : kMPNormalRate;
+    let mpTicks = [
+      baseTick,
+      baseTick + kMPUI1Rate,
```

Comment:
The umbral ice quantity is in jobDetail for black mage, if you want to store it.  Would that make this code easier, so you can just know what the rate is supposed to be?